Core/Types/GameComparator/StrictGameComparator.cs

```diff
@@ -51,8 +52,24 @@ public bool Compatible(KspVersion gameVersion, CkanModule module)
             {
                 return true;
             }
+            
+            if (!moduleRange.Upper.Value.IsAny && isBoundLower(moduleRange.Upper, gameVersionRange.Lower))
+            {
+                return false;
+            }
+
+            if (!moduleRange.Lower.Value.IsAny && isBoundLower(gameVersionRange.Upper, moduleRange.Lower))
+            {
+                return false;
+            }
+
+            return true;
+        }
 
-            return moduleRange.IsSupersetOf(gameVersionRange);
```

Comment:
I see, so what we really want to test is it the ranges intersect each other. I've added such a method in #1958. It returns a new `KspVersionRange` which is equal to the intersection or `null` if there is no intersection. Feel free to cherry-pick or merge in.